pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -168,6 +173,16 @@ public synchronized void start() {
     _helixAdmin = _helixZkManager.getClusterManagmentTool();
     _propertyStore = _helixZkManager.getHelixPropertyStore();
     _helixDataAccessor = _helixZkManager.getHelixDataAccessor();
+    // Cache instance zk paths.
+    BaseDataAccessor<ZNRecord> baseDataAccessor = _helixDataAccessor.getBaseDataAccessor();
+    if (baseDataAccessor instanceof ZkBaseDataAccessor) {
+      String instanceConfigs = PropertyPathBuilder.instanceConfig(_helixClusterName);
+      _cacheInstanceConfigsDataAccessor =
+          new ZkCacheBaseDataAccessor<>((ZkBaseDataAccessor<ZNRecord>) baseDataAccessor, instanceConfigs,
+              Collections.singletonList(instanceConfigs), Collections.singletonList(instanceConfigs));
```

Comment:
If you already give root path. You should not put INSTANCE path again here.  Or if you know that instances are stable, which means there are no instances added or shutdown, you can directly subscribe the paths without root path.

You can refer Helix ZkHelixPropertyStore or this constructor:

  public ZkCacheBaseDataAccessor(ZkBaseDataAccessor<T> baseAccessor, List<String> wtCachePaths) {
    this(baseAccessor, null, wtCachePaths, null);
  }

Reply:
@dasahcc Do you mean keeping one `Collections.singletonList(instanceConfigs)` is enough if I've already given root path?
We can't guarantee instances are stable; we may add/remove instances. But the purpose of getting instance configs is to get the admin endpoint, which consists of hostname and admin port. And these two elements won't be changed often for each instance. So I guess the correct usage of that class is:
```
_cacheInstanceConfigsDataAccessor =
          new ZkCacheBaseDataAccessor<>((ZkBaseDataAccessor<ZNRecord>) baseDataAccessor, instanceConfigs,
              Collections.singletonList(instanceConfigs), null);
```
Could you confirm?

Reply:
That should work. If possible, you can local start a test and modify the instance config through zooinspector to see whether it got updated at cache side.

Reply:
Great! I've tested it using your suggestion and it works perfectly. Thanks for the help! 👍